### PR TITLE
[Opt](Compression) Opt zstd block decompression by `ZSTD_decompressDCtx()` to replace streaming decompression.

### PR DESCRIPTION
**Pull request by kaka11chen**
## Proposed changes

Opt zstd block decompression by `ZSTD_decompressDCtx()` to replace streaming decompression.
It will improve performance but consume more memory. 

Test result: 
- env: 1 node(16 cores, 64G).
- parquet column: 100 million rows of char(255) column.
- result: 5.2 -> 4.6.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/block_compression.cpp [hunk 1]
```diff
@@ -816,38 +816,24 @@ class ZstdBlockCompression : public BlockCompressionCodec {
         return Status::OK();
     }
 
-    // follow ZSTD official example
-    //  https://github.com/facebook/zstd/blob/dev/examples/streaming_decompression.c
```
Comment (yiguolei) [inline be/src/util/block_compression.cpp:820]: add  some sql based regression test using zstd as compression method to ensure that the alg is right.

Comment (kaka11chen) [reply be/src/util/block_compression.cpp:820]: ok, I will add a parquet zstd table regression test later.

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buidall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by AshinGau — APPROVED** [review]

**Pull request merged**